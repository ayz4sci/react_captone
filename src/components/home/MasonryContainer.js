import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Masonry, {ResponsiveMasonry} from "react-responsive-masonry"
import CategoryCard from '../category/CategoryCard';

class MasonryContainer extends Component {

  render() {
    return (
      <ResponsiveMasonry 
        columnsCountBreakPoints={{350: 1, 550: 2, 750: 3, 900: 3, 1024:4}}>
        <Masonry gutter='10'>
            { this.props.categories.map( (item, key) => 
              <CategoryCard category={item}  key={key}/>
            )}
        </Masonry>
      </ResponsiveMasonry>
    );
  }
}

MasonryContainer.protoTypes = {
  categories: PropTypes.array.isRequired
}

export default MasonryContainer;
