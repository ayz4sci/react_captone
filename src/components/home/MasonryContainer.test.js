import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import MasonryContainer from './MasonryContainer';

it('renders correctly', () => {
    const props = { 
        categories: [{
            name: "lol",
            cheats: [{
                description: "lola",
                command: "lola la"
            }]
        }]
    };

    const wrapper = shallow(<MasonryContainer {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot();
});