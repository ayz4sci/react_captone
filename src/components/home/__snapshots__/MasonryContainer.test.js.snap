// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MasonryResponsive
  className={null}
  columnsCountBreakPoints={
    Object {
      "1024": 4,
      "350": 1,
      "550": 2,
      "750": 3,
      "900": 3,
    }
  }
  style={null}
>
  <Masonry
    className={null}
    columnsCount={3}
    gutter="10"
    style={Object {}}
  >
    <CategoryCard
      category={
        Object {
          "cheats": Array [
            Object {
              "command": "lola la",
              "description": "lola",
            },
          ],
          "name": "lol",
        }
      }
      key="0"
    />
  </Masonry>
</MasonryResponsive>
`;
